resources-service/scripts/aselo-resource.ts

```diff
@@ -0,0 +1,40 @@
+export type InlineAttributeTable =
+  | 'ResourceStringAttributes'
+  | 'ResourceBooleanAttributes'
+  | 'ResourceNumberAttributes'
+  | 'ResourceDateAttributes';
+
+export type AseloInlineResourceAttribute<T extends InlineAttributeTable> = {
+  key: string;
+  value: AttributeValue<T>;
+  language: string;
+  info: Record<string, any> | null;
+};
+
+export type AseloResource = {
+  id: string;
+  name: string;
+  attributes: {
+    ResourceStringAttributes: AseloInlineResourceAttribute<'ResourceStringAttributes'>[];
+    ResourceReferenceStringAttributes: {
+      key: string;
+      value: string;
+      language: string;
+      info: Record<string, any> | null;
+      list: string;
+    }[];
+    ResourceBooleanAttributes: AseloInlineResourceAttribute<'ResourceBooleanAttributes'>[];
+    ResourceNumberAttributes: AseloInlineResourceAttribute<'ResourceNumberAttributes'>[];
+    ResourceDateAttributes: AseloInlineResourceAttribute<'ResourceDateAttributes'>[];
```

Comment:
It's not clear to me why instead of:
- `AseloInlineResourceAttribute<'ResourceStringAttributes'>[]`
- `AseloInlineResourceAttribute<'ResourceBooleanAttributes'>[]`
- `AseloInlineResourceAttribute<'ResourceNumberAttributes'>[]`
- `AseloInlineResourceAttribute<'ResourceDateAttributes'>[]`

... we just don't use the `type` directly:
- `AseloInlineResourceAttribute<string>[]`
- `AseloInlineResourceAttribute<boolean>[]`
- `AseloInlineResourceAttribute<number>[]`
- `AseloInlineResourceAttribute<Date>[]`

Reply:
At the moment, you're right, there isn't much difference as they all end up in the string table anyway, but we'll probably put booleans, numbers & dates in their own tables soon, and that type structure allows us to specify the table name, and we infer the type it takes from that

We could set up the mapping the other way around, with the table being inferred from the payload type, but this way seemed better in case we need to split values with the same TS type into 2 different tables (like with reference strings & inline strings, although those are handled separately)